flaskapp/app/routes.py

```diff
@@ -116,96 +118,121 @@ def callback():
         
 ## Functions
 
-def get_venue_info(venue_name=None, venue_city=None, venue_id=None):
-    """Search for venue and retrieve info from Songkick.
-    To handle duplicate venue names, this function will check that venue's city matches that entered in venues.csv""" 
+def seatgeek_request(endpoint, venue_id = None, params = None):
+    """Format a request to SeatGeek API"""
+    parameters = {'client_id': SEATGEEK_CLIENT_ID,
+                 'client_secret': SEATGEEK_CLIENT_SECRET}
+    if params is not None:
+        for key, value in params.items():
+            parameters[key] = value
+    if venue_id is not None and endpoint == 'venues':
+        request_base = f'{SEATGEEK_API_BASE}/{endpoint}/{venue_id}'
+    elif venue_id is not None and endpoint == 'events':
+        request_base = f'{SEATGEEK_API_BASE}/{endpoint}'
+        parameters['venue.id'] = venue_id
+    else:
+        request_base = f'{SEATGEEK_API_BASE}/{endpoint}'
+    response = requests.get(request_base, params=parameters)
+    return response
+
+def get_venue_info(venue_name = None, venue_city = "", venue_id = None):
```

Comment:
Oh I see, you kinda already have that. Honestly we're probably getting to the point where breaking up this file into a more abstract - but ultimately organized - structure of services, models, etc. is going to be useful. That would help us break up the function above without feeling like we're adding a bunch of new functions to this already long file. I'd say let's chat this over next time we talk.

Reply:
That would be super helpful! Restructuring is something I've had on my to-do list for a long time.